File: remote/src/main/java/com/kelsos/mbrc/presenters/BrowseGenrePresenter.java
package com.kelsos.mbrc.presenters;

import com.kelsos.mbrc.annotations.Queue;
import com.kelsos.mbrc.domain.Genre;
import com.kelsos.mbrc.ui.views.BrowseGenreView;

public interface BrowseGenrePresenter {
  void bind(BrowseGenreView view);

  void load();

  void queue(Genre genre, @Queue.Action String action);
}

File: remote/src/main/kotlin/com/kelsos/mbrc/presenters/BrowseGenrePresenter.kt
package com.kelsos.mbrc.presenters

import com.kelsos.mbrc.annotations.Queue
import com.kelsos.mbrc.domain.Genre
import com.kelsos.mbrc.ui.views.BrowseGenreView

interface BrowseGenrePresenter : PlaylistAdder {
  fun bind(view: BrowseGenreView)

  fun load()

  fun queue(genre: Genre, @Queue.Action action: String)

  fun load(page: Int)
}
